src/pages/MarkingPage/Scrollspy/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { Flex } from '@radix-ui/themes'
+import { isEmpty, isNil, sumBy } from 'lodash'
+import React, { FC } from 'react'
+
+import { Question as QuestionType } from '../../../types/exam'
+import { MarkRoot } from '../../../types/marking'
+import { numberToLetter, numberToRoman } from '../../../utils/common'
+import { ScrollspyItem } from './ScrollspyItem'
+
+interface ScrollspyProps {
+  questions: Record<number, QuestionType>
+  marks: MarkRoot[]
+  activeId: string | undefined
+}
+
+const Scrollspy: FC<ScrollspyProps> = ({ questions, marks, activeId }) => {
+  function questionPartial(q: number): number | undefined {
+    let relevantMarks = marks.filter((m) => m.question === q && !isNil(m.mark))
+    return isEmpty(relevantMarks) ? undefined : sumBy(relevantMarks, 'mark')
+  }
+
+  function sectionPartial(q: number, p: number, s: number): number | undefined {
```

Comment:
Similar, maybe change name to `marksForSection`?

Reply:
I am going to settle for `currentSectionMark`